core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -19,9 +19,16 @@
 
 oppia.factory('UrlService', ['$window', function($window) {
   return {
+    // This function is for testing purposes (to mock $window.location)
+    queryUrlParams: function() {
+      return $window.location;
+    },
+    /* As params[key] is overwritten, if query string has multiple fieldValues
+       for same fieldName, use getQueryFieldValuesAsList(fieldName) to get it
+       in array form. */
     getUrlParams: function() {
       var params = {};
-      var parts = $window.location.href.replace(
+      var parts = this.queryUrlParams().href.replace(
```

Comment:
Since you use this.queryUrlParams().href a lot, maybe pull that out into getCurrentHref()

Reply:
Ok, I did that, but is it ok if I called getCurrentLocation() inside this?

Reply:
Yep! In fact it's preferable.